src/ngraph/pass/manager.cpp

```diff
@@ -98,6 +99,10 @@ void ngraph::pass::Manager::run_passes(shared_ptr<Function> func, bool transitiv
         }
         else if (function_pass)
         {
+            if (auto pc_pass = dynamic_pointer_cast<pass::PropagateCacheability>(function_pass))
+            {
+                pc_pass->set_op_annotations_func(get_state().get_op_annotations_func());
```

Comment:
can we come up with a better name? `set_annotation_factory`, `set_annotation_creator`, `set_annotation_producer`? @jbobba

Reply:
set_annotation_factory